File: day7/day7.py
def containsBag(name, insideBags):
    for child in insideBags:
        if name == child:
            return True
        if containsBag(name,data[child]):
            return True
    return False
def countInsideBags(insideBags):
    if insideBags == {}:
        return 0
    total = 0
    for insideBagName in insideBags:
        total += ((countInsideBags(data[insideBagName]) + 1) * insideBags[insideBagName])
    return total

f = open("day7.txt","r")
bag_list = f.readlines()
data = {}
for line in bag_list:
    words = line.split()
    key = words[0] + ' ' + words[1]
    data[key] = {}
    for i, word in enumerate(words):
        if words[i].isnumeric():
            subKey = words[i + 1] + ' ' + words[i + 2]
            data[key][subKey] = int(word)
print(data)

answer = []
searchName = "shiny gold"

for bag in data:
    if containsBag(searchName, data[bag]):
        answer.append(bag)
#part1
print(len(answer))
#part2
print(countInsideBags(data[searchName]))

File: day7/day7.js
const fs = require('fs')

fs.readFile('./day7.txt', 'utf8' , (err, data) => {
    //console.log(bags);
    part1 = 0;
    bags = data.split("\r\n");
    //console.log(bags)
    var find = "shiny gold bag"
    var dict = {};
    bags.forEach(element => {
        bag_parts = element.split(" contain ")
        //console.log(bag_parts)
        main_bag = bag_parts[0]
        dict[main_bag] = []
        inside_bags = bag_parts[1].split(", ");
        inside_bags.forEach(element => {
            dict[main_bag].push(element);
        });
    });

    set = new Set();
    for (var key in dict) {
        for (var i = 0; i < dict[key].length; i++) {

            if(dict[key][i].includes(find) && set.){
                part1 +=1
                set.add(dict[key][i]);
            }
            else {
                //get the bags contained in this bag
            }
        }
    }
    console.log(part1);
})
//key will be bag name, values will be bag name